src/rgw/rgw_op.cc

```diff
@@ -488,8 +488,15 @@ static int read_obj_policy(const DoutPrefixProvider *dpp,
     const rgw_user& bucket_owner = bucket_policy.get_owner().get_id();
     if (bucket_owner.compare(s->user->get_id()) != 0 &&
         ! s->auth.identity->is_admin_of(bucket_owner)) {
+      for (const auto& policy : s->iam_user_policies) {
```

Comment:
This name ends up shadowing the older `policy` variable. Maybe use a distinct name here?